<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1207</width>
    <height>699</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>SEM Particle Counter</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <widget class="QFrame" name="frame">
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <layout class="QGridLayout" name="gridLayout_15">
       <item row="0" column="0">
        <widget class="QGroupBox" name="groupBox">
         <property name="title">
          <string>Input file selection</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_3">
          <item row="0" column="0">
           <layout class="QGridLayout" name="gridLayout_2">
            <item row="1" column="1">
             <widget class="QPushButton" name="pushButton_nextimage">
              <property name="toolTip">
               <string>Open next .tif in folder</string>
              </property>
              <property name="text">
               <string>Next</string>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QPushButton" name="pushButton_previousimage">
              <property name="toolTip">
               <string>Open previous .tif in folder</string>
              </property>
              <property name="text">
               <string>Previous</string>
              </property>
             </widget>
            </item>
            <item row="0" column="0" colspan="2">
             <widget class="QPushButton" name="pushButton_fileinput">
              <property name="toolTip">
               <string>Choose an SEM .tif file to analyse</string>
              </property>
              <property name="text">
               <string>Select image file</string>
              </property>
             </widget>
            </item>
            <item row="2" column="0" colspan="3">
             <widget class="QLabel" name="label_inputname">
              <property name="text">
               <string>(No file currently selected)</string>
              </property>
              <property name="wordWrap">
               <bool>true</bool>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QGroupBox" name="groupBox_2">
         <property name="title">
          <string>Image calibration</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_5">
          <item row="0" column="0">
           <layout class="QGridLayout" name="gridLayout_4">
            <item row="0" column="3">
             <widget class="QPushButton" name="pushButton_updatescalenm">
              <property name="enabled">
               <bool>false</bool>
              </property>
              <property name="toolTip">
               <string>Update calibration based on input scalebar size</string>
              </property>
              <property name="text">
               <string>Update</string>
              </property>
             </widget>
            </item>
            <item row="0" column="0" colspan="2">
             <widget class="QLabel" name="label">
              <property name="text">
               <string>Scalebar size / nm</string>
              </property>
             </widget>
            </item>
            <item row="0" column="2">
             <widget class="QLineEdit" name="lineEdit_scalenm">
              <property name="enabled">
               <bool>false</bool>
              </property>
              <property name="toolTip">
               <string>Image scalebar length in nm (used for image calibration)</string>
              </property>
             </widget>
            </item>
            <item row="1" column="0" colspan="4">
             <widget class="QLabel" name="label_pxwidth">
              <property name="text">
               <string>Pixel width:</string>
              </property>
             </widget>
            </item>
            <item row="2" column="0" colspan="4">
             <widget class="QLabel" name="label_imdim">
              <property name="text">
               <string>Image dimensions:</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item row="0" column="2">
        <widget class="QGroupBox" name="groupBox_5">
         <property name="acceptDrops">
          <bool>false</bool>
         </property>
         <property name="title">
          <string>Automatic identification</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_9">
          <item row="0" column="2">
           <layout class="QGridLayout" name="gridLayout_8">
            <item row="1" column="0" colspan="2">
             <widget class="QCheckBox" name="checkBox_f2">
              <property name="toolTip">
               <string>Attempts to find shapes within clusters by looking for edges</string>
              </property>
              <property name="text">
               <string>Edge detection</string>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item row="3" column="0" colspan="2">
             <widget class="QCheckBox" name="checkBox_f4">
              <property name="enabled">
               <bool>false</bool>
              </property>
              <property name="text">
               <string>Filter 4</string>
              </property>
              <property name="checkable">
               <bool>false</bool>
              </property>
             </widget>
            </item>
            <item row="2" column="0" colspan="2">
             <widget class="QCheckBox" name="checkBox_f3">
              <property name="enabled">
               <bool>false</bool>
              </property>
              <property name="text">
               <string>Filter 3</string>
              </property>
              <property name="checkable">
               <bool>false</bool>
              </property>
             </widget>
            </item>
            <item row="4" column="0" colspan="2">
             <widget class="QCheckBox" name="checkBox_f5">
              <property name="enabled">
               <bool>false</bool>
              </property>
              <property name="text">
               <string>Filter 5</string>
              </property>
              <property name="checkable">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item row="0" column="0" colspan="2">
             <widget class="QCheckBox" name="checkBox_f1">
              <property name="toolTip">
               <string>Good at finding cluster outlines</string>
              </property>
              <property name="text">
               <string>Simple threshold</string>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item row="0" column="3">
           <layout class="QGridLayout" name="gridLayout_7">
            <item row="1" column="0">
             <widget class="QLabel" name="label_4">
              <property name="text">
               <string>Max size / nm</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QSpinBox" name="spinBox_minsize">
              <property name="toolTip">
               <string>Minimum particle size boundary</string>
              </property>
              <property name="maximum">
               <number>10000</number>
              </property>
              <property name="value">
               <number>30</number>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QSpinBox" name="spinBox_maxsize">
              <property name="toolTip">
               <string>Maximum particle size boundary</string>
              </property>
              <property name="maximum">
               <number>10000</number>
              </property>
              <property name="value">
               <number>1000</number>
              </property>
             </widget>
            </item>
            <item row="0" column="0">
             <widget class="QLabel" name="label_3">
              <property name="text">
               <string>Min size / nm</string>
              </property>
             </widget>
            </item>
            <item row="2" column="0" colspan="2">
             <widget class="QPushButton" name="pushButton_autoidentify">
              <property name="text">
               <string>Identify shapes</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item row="0" column="3">
        <widget class="QGroupBox" name="groupBox_6">
         <property name="title">
          <string>Manual identification</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_10">
          <item row="1" column="0">
           <widget class="QLabel" name="label_9">
            <property name="text">
             <string>Left click = Add point</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="label_11">
            <property name="text">
             <string>Right click = Finish</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label_10">
            <property name="text">
             <string>Middle click = Remove point</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QPushButton" name="pushButton_manualidentify">
            <property name="toolTip">
             <string>Add a single shape manually. 
 Left click to add points. 
 Middle click to remove points. 
 Right click to finish. 
 Note timout after 60 s)</string>
            </property>
            <property name="text">
             <string>Add shape</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item row="0" column="4">
        <spacer name="horizontalSpacer_3">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>123</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="0" column="5">
        <widget class="QGroupBox" name="groupBox_7">
         <property name="title">
          <string>Output</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_11">
          <item row="0" column="0">
           <widget class="QPushButton" name="pushButton_outputfile">
            <property name="toolTip">
             <string>Choose output file for saving. 
 If creating a new file, the program will create 'userinput_HEX.csv' and 'userinput_ROD.csv'. 
 If file already exists, select either the HEX or ROD file and the program will automatically add to them.</string>
            </property>
            <property name="text">
             <string>Select/create output file</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_output_path">
            <property name="text">
             <string>(No file currently selected)</string>
            </property>
            <property name="wordWrap">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QPushButton" name="pushButton_addtofile">
            <property name="toolTip">
             <string>Saves everything in 'current shapes' to the selected output files</string>
            </property>
            <property name="text">
             <string>Add current shapes to file</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item row="1" column="0" colspan="5">
        <widget class="QGroupBox" name="groupBox_3">
         <property name="title">
          <string>Current image</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_6">
          <item row="1" column="1">
           <spacer name="verticalSpacer">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
          <item row="0" column="0">
           <spacer name="horizontalSpacer">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item row="1" column="0">
           <layout class="QVBoxLayout" name="verticalLayout_plotwindow"/>
          </item>
         </layout>
        </widget>
       </item>
       <item row="1" column="5">
        <widget class="QGroupBox" name="groupBox_4">
         <property name="title">
          <string>Current shapes</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_12">
          <item row="2" column="0">
           <widget class="QScrollArea" name="scrollArea">
            <property name="widgetResizable">
             <bool>true</bool>
            </property>
            <widget class="QWidget" name="scrollAreaWidgetContents">
             <property name="geometry">
              <rect>
               <x>0</x>
               <y>0</y>
               <width>131</width>
               <height>331</height>
              </rect>
             </property>
             <layout class="QGridLayout" name="gridLayout_14">
              <item row="1" column="0">
               <spacer name="verticalSpacer_2">
                <property name="orientation">
                 <enum>Qt::Vertical</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>20</width>
                  <height>40</height>
                 </size>
                </property>
               </spacer>
              </item>
              <item row="0" column="0">
               <layout class="QGridLayout" name="gridLayout_13">
                <item row="0" column="2">
                 <widget class="QLabel" name="label_7">
                  <property name="text">
                   <string>Rod</string>
                  </property>
                 </widget>
                </item>
                <item row="0" column="1">
                 <widget class="QLabel" name="label_5">
                  <property name="text">
                   <string>Hex</string>
                  </property>
                 </widget>
                </item>
                <item row="0" column="0">
                 <widget class="QLabel" name="label_2">
                  <property name="text">
                   <string>n</string>
                  </property>
                 </widget>
                </item>
                <item row="0" column="3">
                 <widget class="QLabel" name="label_8">
                  <property name="text">
                   <string>Keep?</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
             </layout>
            </widget>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QPushButton" name="pushButton_updateshapes">
            <property name="toolTip">
             <string>Removes any shapes with 'Keep?' unchecked</string>
            </property>
            <property name="text">
             <string>Update shapes</string>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QCheckBox" name="checkBox_shapetoggle">
            <property name="toolTip">
             <string>Toggle plotting of shapes on/off</string>
            </property>
            <property name="text">
             <string>Toggle shape view</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QCheckBox" name="checkBox_selectall">
            <property name="toolTip">
             <string>Toggle all 'Keep?' checkboxes on/off</string>
            </property>
            <property name="text">
             <string>(De)Select all</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="label_statusbar">
         <property name="text">
          <string>Status: Idle</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1207</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
